src/Chapter2.hs

```diff
@@ -606,7 +612,8 @@ Implement a function that duplicates each element of the list
 
 -}
 duplicate :: [a] -> [a]
-duplicate = error "duplicate: Not implemented!"
+duplicate [] = []
+duplicate (x:xs) = replicate 2 x ++ duplicate xs
```

Comment:
This is right 🙂 

However, for 2 elements it is possible to write it by hands without usage of `++` operator to make it a bit faster:

```suggestion
duplicate (x:xs) = x : x : duplicate xs
```